lightdm: allow `dm-tool lock` to work properly

philandstuff opened the pull request:
Fixes #5038.

The `dm-tool lock` command fails to work for reasons described in #5038:
it tries passing arguments to the xserver command, but the wrapper
throws the arguments away and replaces them with different arguments.

I fixed this by specifying the xserver args in the lightdm.conf file and
having the wrapper pass them through.  This works for lightdm on
startup, and also works for `dm-tool lock` when it later passes a
different set of arguments to start a separate X server on another
display.

Ping @ocharles @wkennington, who are listed as maintainers of the lightdm package.

philandstuff committed:
lightdm: allow `dm-tool lock` to work properly

Fixes #5038.

The `dm-tool lock` command fails to work for reasons described in #5038:
it tries passing arguments to the xserver command, but the wrapper
throws the arguments away and replaces them with different arguments.

I fixed this by specifying the xserver args in the lightdm.conf file
instead of in the wrapper.  This works for lightdm on startup, and also
works for `dm-tool lock` when it later passes a different set of
arguments to start a separate X server on another display.

philandstuff commented:
It's probably worth saying that `dm-tool lock` [doesn't, by itself, lock the screen](https://bugs.launchpad.net/ubuntu/+source/lxsession/+bug/1205384) so there's still more work to be done after this PR.

ocharles commented:
This looks OK from me, @viric, @wkennington - any thoughts?

zimbatm commented:
@philandstuff do you mind rebasing on top of master ?

There is just one thing that needs to be fixed: $@ needs to be quoted for shell expansion to work properly. So `"$@"` instead of just `$@`

philandstuff commented:
hmm, it looks like 9be012f0d4d9b6bc by @obadz (merged in #11338) has already added the `"$@"` pass-through. It certainly looks like it was solving a related problem.

The only thing remaining in this PR is whether `${dmcfg.xserverArgs}` should remain hardcoded in the xserver-wrapper script, or if  put in to lightdm.conf.  I kind of feel that putting the xserverArgs in lightdm.conf makes some of the logic in the xserver-wrapper unnecessary, but since I opened this PR I've moved to gdm instead of lightdm and so I don't have the setup to test this change any more.  #11338 changed the semantics of xserverArgs so I don't have confidence in this.

I've rebased the commit and I'll leave this open for a few days if someone wants to take it up, but if there's no further activity within a week I suggest closing this PR.

As an aside, I wonder if #5038 was fixed by #11338?

obadz commented:
@philandstuff I wonder that too

obadz commented:
@philandstuff actually I'm pretty sure it's fixed

philandstuff commented:
Closing (re https://github.com/NixOS/nixpkgs/pull/11686#issuecomment-189627372 )
